tsconfig.json

```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     // Enable top-level await, and other modern ESM features.
     "target": "ESNext",
-    "module": "es2022",
```

Comment:
It appears that Hardhat doesn't support ESM in TypeScript projects. According to the [Hardhat documentation](https://hardhat.org/hardhat-runner/docs/advanced/using-esm), we need to either leave the `module` as "CommonJS" or if we want to use ESM by default, we have to use a JavaScript configuration file for Hardhat by naming it `hardhat.config.cjs` (instead of `hardhat.config.ts`).